src/rest_framework_reactive/consumers.py

```diff
@@ -6,12 +6,37 @@
 from channels.consumer import AsyncConsumer, SyncConsumer
 from channels.generic.websocket import JsonWebsocketConsumer
 
+from django.core.cache import cache
+
+from .connection import get_queryobserver_settings
 from .models import Observer, Subscriber
 from .observer import QueryObserver
 from .protocol import *
 
 # Maximum number of cached observer executors.
 MAX_CACHED_EXECUTORS = 1024
+# Waiting observers cache name
+WAITING_OBSERVERS_CACHE_PREFIX = 'drf_reactive_observer_throttle_'
+
+
+def throttle_semaphore(observer_id):
+    """Check if observer should be evaluated, delayed or ignored.
+
+    Increase the observer counter if throttle cache exists.
+    """
+    cache_key = '{}{}'.format(WAITING_OBSERVERS_CACHE_PREFIX, observer_id)
+    throttle_rate = get_queryobserver_settings()['throttle_rate']
+
+    try:
+        count = cache.incr(cache_key)
+        # Ignore if delayed observer already scheduled.
+        return 'delay' if count == 2 else 'ignore'
+    except ValueError:
+        count = cache.get_or_set(cache_key, default=1, timeout=throttle_rate)
+        # Ignore if cache was set and increased in another thread.
+        return 'evaluate' if count == 1 else 'ignore'
```

Comment:
This will return `evaluate` almost always, as it will be `1` even if the cache is set by another process.